lib/chef/chef_fs/knife.rb

```diff
@@ -98,14 +99,41 @@ def pattern_args_from(args)
       end
 
       def pattern_arg_from(arg)
-        # TODO support absolute file paths and not just patterns?  Too much?
-        # Could be super useful in a world with multiple repo paths
-        if !@chef_fs_config.base_path && !Chef::ChefFS::PathUtils.is_absolute?(arg)
-          # Check if chef repo path is specified to give a better error message
-          ui.error("Attempt to use relative path '#{arg}' when current directory is outside the repository path")
+        inferred_path = nil
+        if Chef::ChefFS::PathUtils.is_absolute?(arg)
```

Comment:
@jkeiser This is for you!

C:\REPO\KSUBRAMA\cHeF-intermediate [master]> bundle exec knife upload C:\repo\ksubrama\chef-intermediate\cookbooks\apache -VV
DL is deprecated, please use Fiddle
INFO: Using configuration from C:/REPO/KSUBRAMA/cHeF-intermediate/.chef/knife.rb
WARNING: An absolute file system path that isn't a server path was provided: C:\repo\ksubrama\chef-intermediate\cookbooks\apache
WARNING: Paths provided need to be rooted at the chef-repo being considered or be relative paths.
WARNING: Using '/cookbooks/apache' as the path instead of 'C:\repo\ksubrama\chef-intermediate\cookbooks\apache'.